progenitor-impl/src/method.rs

```diff
@@ -2203,8 +2232,101 @@ impl Generator {
                         content_type, schema
                     ))),
                 }?;
+
                 OperationParameterType::RawBody
             }
+            BodyContentType::FormData => {
+                // For form data, we expect a key-value set of types, specific schema:
+
+                // ```yaml
+                //    type: "object"
+                //      properties:
+                //        file:
+                //          description: "The file to upload"
+                //          type: "string"
+                //          format: "binary"
+                // ```
+                // "schema": {
+                //     "type": "string",
+                //     "format": "binary"
+                // }
+
+                let _mapped = match schema.item(components)? {
+                    openapiv3::Schema {
+                        schema_data:
+                            openapiv3::SchemaData {
+                                nullable: false,
+                                discriminator: None,
+                                default: None,
+                                // Other fields that describe or document the
+                                // schema are fine.
+                                ..
+                            },
+                        schema_kind:
+                            openapiv3::SchemaKind::Type(openapiv3::Type::Object(
+                                openapiv3::ObjectType {
+                                    properties,
+                                    additional_properties: _,
+                                    ..
+                                },
+                            )),
+                    } => {
+                        let mapped = Result::<IndexSet<_>>::from_iter(
+                            properties
+                                .into_iter()
+                                .map(|(name, property)| {
+                                        // properties must be plain key value types for now 
+                                        let ReferenceOr::Item(property) = property else {
+                                                return Err(Error::UnexpectedFormat(format!(
+                                                    "invalid schema for {}: didn't expect a reference",
+                                                    BodyContentType::FormData,
+                                                )))};
+                                        match &property.schema_kind {
+                                            openapiv3::SchemaKind::Type(openapiv3::Type::String(
+                                                openapiv3::StringType {
+                                                    format:
+                                                        openapiv3::VariantOrUnknownOrEmpty::Item(
+                                                            openapiv3::StringFormat::Binary,
```

Comment:
Are you planning to get rid of this limitation as part of this PR? My use-case requires support for `boolean`s and `string`s with format `date-time`, `uuid` or no specified format.

I reviewed your code, but I don't think I'd be able to add support for them without some guidance :thinking:

Reply:
I think it's there be cause my specific use case only required binary.

Could you provide a minimal/trimmed down openapi spec where this'd fail?

Reply:
Sure, here's a trimmed down spec I'm using: [example.json](https://github.com/user-attachments/files/16515930/example.json)

This is based on the OpenAPI spec for https://github.com/immich-app/immich